src/api/fetch.js

```diff
@@ -35,101 +38,89 @@ export async function* indexGenerator(path, auth) {
   }
 }
 
-export function create(path, auth, object) {
-  return fetch(`${baseURL}/${path}`, {
+async function resolveRequest(request) {
+  let response;
+  try {
+    response = await fetch(request);
+  } catch (reason) {
+    throw { error: [reason] };
+  }
+  const result = await response.json();
+  if (response.ok) {
+    return result;
+  } else {
+    throw result;
+  }
+}
```

Comment:
```suggestion
  try {
    const response = await fetch(request);
    const result = await response.json();
    if (response.ok) {
      return result;
    } else {
      throw result;
    }
  } catch (reason) {
    throw { error: [reason] };
  }
}
```
This is a bit cleaner, IMO.

Reply:
This doesn't work.
If we have an error from the server (say an album without title) we will throw iit on line 48 (your version), then catch it and wrap it in an error object on line 51. As a result our errors don't display correctly anymore (but as `errors.[object object]`)

Reply:
You're right. Ugh, that's ugly, but fine.